uiowa/uiowa
# UI Press - Book migrations

Resolves #4662 

# How to test

Sync uipress in DevDesktop (grad container)
git checkout uipress_book_migration && git pull
blt frontend
blt ds --site=uipress.uiowa.edu
drush @uipress.local en uipress_migrate -y
drush @uipress.local uli /admin/config/sitenow/sitenow-migrate
Complete the form using the help text
drush @uipress.local ms <- check that migrations register
drush @uipress.local mim --all

## Books to compare:

Compare "[Zig-Zag-and-Swirl](https://www.uipress.uiowa.edu/books/9781587294662/zig-zag-and-swirl)" for author, subjects, book cover, publication date (2010 year), pages/art/..., book types (paper, ebook 120 and perpetual), summary, reviews/blurbs, related
Compare "[They Don’t Want Her There](https://www.uipress.uiowa.edu/books/9781609388195/they-don%E2%80%99t-want-her-there)" for foreword author, afterword author
Compare "[Women Adapting](https://www.uipress.uiowa.edu/books/9781609386498/women-adapting)" for series, awards
Compare "[Whitman & Dickinson](https://www.uipress.uiowa.edu/books/9781609385316/whitman-dickinson)" for editors, contributors
Compare "[The Raptors of Iowa](https://www.uipress.uiowa.edu/books/9781609381660/the-raptors-of-iowa)" for illustrator, essay, table of contents
Compare "[Vivid and Continuous](https://www.uipress.uiowa.edu/books/9781609381561/vivid-and-continuous)" for excerpt
Compare "[Wildflowers of Iowa Woodlands](https://www.uipress.uiowa.edu/books/9781587298233/wildflowers-of-iowa-woodlands)" for interior images
Compare '[“The Million Dead, Too, Summ’d Up”](https://www.uipress.uiowa.edu/books/9781609387464/%E2%80%9Cthe-million-dead-too-summ%E2%80%99d-up%E2%80%9D)' for introductory author
Compare "[Turtles in Your Pocket](https://www.uipress.uiowa.edu/books/9781609380618/turtles-in-your-pocket)" for photographer
Compare "[Poems from Guantánamo](https://www.uipress.uiowa.edu/books/9781587296062/poems-from-guant%C3%A1namo)" for preface authors
Compare "[Memoirs of a Revolutionary](https://www.uipress.uiowa.edu/books/9780877458272/memoirs-of-a-revolutionary)" for translator

## Note

There will be 38 "stub" nodes in the top of the content overview. Other nodes referenced this content, the migration stubbed it out, but ultimately the content to fill these out no longer exists. Manual deletion of these should be done after migration. When the referencing nodes are saved again, those references should be cleaned up.
